apps/web/src/views/Farms/v4/components/PoolListView.tsx

```diff
@@ -0,0 +1,202 @@
+import { TokenPairImage } from 'components/TokenImage'
+import { TokenOverview } from '@pancakeswap/widgets-internal'
+import { PoolInfo } from 'state/farmsV4/state/type'
+import styled from 'styled-components'
+import { useTranslation } from '@pancakeswap/localization'
+import { BottomDrawer, Column, Button, MoreIcon, ChevronRightIcon } from '@pancakeswap/uikit'
+import { ReactNode, useCallback, useMemo, useState } from 'react'
+import { useColumnMobileConfig } from './useColumnConfig'
+
+const ListContainer = styled.ul``
+
+const ListItemContainer = styled.li`
+  display: flex;
+  padding: 12px 16px;
+  align-items: center;
+  gap: 16px;
+  justify-content: space-between;
+  border-bottom: 1px solid ${({ theme }) => theme.colors.cardBorder};
+  background: 1px solid ${({ theme }) => theme.card.background};
+
+  &:first-child {
+    border-top: 1px solid ${({ theme }) => theme.colors.cardBorder};
+  }
+`
+
+interface IPoolListViewProps {
+  data: PoolInfo[]
+}
+export const ListView: React.FC<IPoolListViewProps> = ({ data }) => {
+  const [openItem, setOpenItem] = useState<PoolInfo | null>(null)
+  const handleDrawerChange = useCallback((status: boolean) => {
+    if (!status) setOpenItem(null)
+  }, [])
+
+  const drawerContent = useMemo(() => <ListItemDetails data={openItem} />, [openItem])
```

Comment:
why no just use  `Memo(ListItemDetails)`

Reply:
yeah, Memo may the best practice

Reply:
fixed